api/v1alpha3/virtualmachine_types.go

```diff
@@ -102,6 +102,10 @@ const (
 	// VirtualMachineReconcilePausedReason indicates that VirtualMachine
 	// reconciliation is being paused.
 	VirtualMachineReconcilePausedReason = "VirtualMachineReconcilePaused"
+
+	// VirtualMachineConditionPrePowerOnReconfigReady exposes the status of the
+	// VM's pre-power-on reconfiguration task.
+	VirtualMachineConditionPrePowerOnReconfigReady = "VirtualMachinePrePowerOnReconfigReady"
```

Comment:
I'm a little uneasy that this condition exposes an internal implementation detail. In a lot of cases we don't need that reconfigure b/c of the 7605 work, and ideally I want to get us to where the VM is created with its desired configuration in all cases. And with resize coming that is going to take the place of the current prepower on reconfigure, and that will have its own conditon.

Reply:
The purpose of this new condition is to inform users when they attempt to deploy a VM with an invalid guestID, as it's no longer verified by a generated list from the imported govmomi package. While implementing this change, I thought why make it specific to guestID only, hence the `PrePowerOnReconfigureReady` name above to expose any VM reconfigure failure (it still has config.guestID in the condition's reason if the task fails by that).

On the other hand, today VMOP already sends a K8s event if the `PrePowerOnReconfigure` task fails. This event contains specific invalid configs as well, such as guestID shown in the following example. So I think users are not entirely in the dark if they have specified an unsupported guestID:
```
Warning  CreateOrUpdateFailure  29s (x16 over 116s)  vmware-system-vmop/vmware-system-vmop-controller-manager-745fcd885b-v2xhm/virtualmachine-controller  reconfigure VM task failed: A specified parameter was not correct: configSpec.guestId
``` 
@akutz Any thoughts on dropping this condition and leveraging existing K8s events for this?

Reply:
I'd prefer to not start exposing prePowerOnReconfigure as a condition too.

Reply:
Updated with the Guest ID specific condition. Please refer to the PR's description for all possible success/failure cases.